src/devices/bus/coco/cococart.cpp

```diff
@@ -443,6 +445,21 @@ device_cococart_interface::~device_cococart_interface()
 
 
 //-------------------------------------------------
+//  interface_config_complete
+//-------------------------------------------------
+
+void device_cococart_interface::interface_config_complete()
+{
+	m_owning_slot = dynamic_cast<cococart_slot_device *>(device().owner());
+	if (!m_owning_slot)
+		throw new emu_fatalerror("Expected device().owner() to be of type cococart_slot_device");
```

Comment:
Your Java/C# strikes again.  Ixnay on the `new` - you're throwing a pointer to an object that will leak, whereas you want to throw an instance of an object.